rust/src/dataset/write.rs

```diff
@@ -115,8 +115,10 @@ pub async fn write_fragments(
     base_dir: &Path,
     schema: &Schema,
     data: SendableRecordBatchStream,
-    params: WriteParams,
+    mut params: WriteParams,
 ) -> Result<Vec<Fragment>> {
+    // Make sure the max rows per group is not larger than the max rows per file
+    params.max_rows_per_group = std::cmp::min(params.max_rows_per_group, params.max_rows_per_file);
```

Comment:
Found this small bug: If you set the `max_rows_per_file` to something smaller than the default `max_rows_per_group`, then it won't split files as expected. So this enforces `max_rows_per_group <= max_rows_per_file`.